### PR TITLE
heal: Adapt the status based on a new Finished flag

### PR DESCRIPTION
**Pull request by vadmeste**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
In a new MinIO version, Healing can be set to false while HealInfo is not nil. 
This indicates that the drive was healed and the healing is finished. This commit
 adapts the code with the new concept. This will not break with older MinIO 
versions since those versions will already remove .healing.bin, in that case 
Healing will be false, HealInfo will be nil.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by harshavardhana** [comment]
update go.mod @vadmeste also fix the title.

**Comment by vadmeste** [comment]
@harshavardhana PTAL

File: go.mod [hunk 1]
```diff
@@ -121,3 +121,5 @@ require (
 	google.golang.org/protobuf v1.34.2 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace github.com/minio/madmin-go/v3 => github.com/vadmeste/madmin-go/v3 v3.0.0-20240812132301-2f1172f311fb
```
Comment (klauspost) [inline go.mod:125]: Isn't it merged so we can get rid of this?

Comment (vadmeste) [reply go.mod]: yes, thanks for catching this

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**